pyhole/core/logger.py

```diff
@@ -14,13 +14,53 @@
 
 """Pyhole Logging"""
 
+import bz2
+import glob
 import logging
 import logging.handlers
+import os
 import requests
+import shutil
 
 import utils
 
 
+LOG_DIR = utils.get_directory("logs")
+LOG_ARCHIVE_DIR = utils.get_directory(os.path.join("logs", "archive"))
+LOG_FORMAT = "%(asctime)s [%(name)s] %(message)s"
+LOG_DATEFMT = "%H:%M:%S"
+
+
+class PyholeFileHandler(logging.handlers.TimedRotatingFileHandler):
+    # CamelCase because superclass is like that
+    def doRollover(self):
+        result = super(PyholeFileHandler, self).doRollover()
+        self.archive_old_logs()
+        return result
+
+    def archive_old_logs(self):
+        # Compress uncompressed logs
+        matcher = "*.log.*[!b][!z][!2]"
+        files = glob.glob(os.path.join(LOG_DIR, matcher))
+        for file_path in files:
+            filename = os.path.basename(file_path)
+            compressed_filename = filename + ".bz2"
+            network_name = filename[:filename.rfind('.log')]
+            archive_dir = utils.get_directory(os.path.join("logs",
```

Comment:
Right now no, utils.get_directory makes the directory but requires a relative path from home. And it returns an absolute path.